Desktop App Automatic & Manual Updates (#5628)
mattermost/docs: source/install/desktop-app-install.rst

- (Linux) `Set up Desktop Entry <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ for the application to more easily `integrate into a desktop environment <https://wiki.archlinux.org/index.php/Desktop_entries>`__.

The following additional documentation resources are also available:

- `Desktop App changelog <https://docs.mattermost.com/install/desktop-app-changelog.html>`__
- `Minimum software requirements <https://docs.mattermost.com/install/software-hardware-requirements.html#desktop-apps>`__
- `Configure your Desktop App experience <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__
- `Source code <https://github.com/mattermost/desktop>`__
- `Contributor’s guide <https://developers.mattermost.com/contribute/desktop>`__

Install the Mattermost Desktop App
----------------------------------

You can `download the Desktop App directly from our Downloads page <https://mattermost.com/download/>`__. You may also use the following installation guides for Linux, Mac, and Windows.

.. include:: ../upgrade/upgrading-to-v60.rst
  :start-after: :nosearch:

.. tabs::

  .. tab:: Linux

    **Generic Linux package** 

>     1. Download the latest version of the Mattermost desktop app:

Author: Need updated download URLs

Reviewer: The ``5.0.4`` can be replaced with ``5.1.0`` in the URLs.
I have a task as part of the release checklists to update the URLs before a release - I'm 0/5 if you update the URLs as part of this PR, or I can update in a separate PR. Up to you!

Author: There may be a URL that takes users to the latest available version. That would be ideal and would eliminate the need to maintain these URLs each release. @devinbinnie - would you be open to investigating this further?

Reviewer 2: I might be mistaken here, but the ones I may have been thinking of are for the Rainforest builds, which are currently pointing to daily, which isn't ideal for our docs.

Other than that I don't know where I would be able to investigate since I don't control those URLs. @amyblais If we don't have this who would be best to set up a link like I described? I think it would save some of @cwarnermm's time so that she doesn't have to update everytime we release an app.

Reviewer: @devinbinnie Normally I have been updating the download links for each release as part of the release checklists. @cwarnermm, let me know if you'd like me to help with this for v5.1.

I'm not sure if @pfltdv knows if we have a "latest" download link for desktop app, or maybe we can create a ticket to add it for future.

Author: @amyblais - Very happy for your help for this upcoming v6.7 release. Longer term, it would be ideal to have a "latest" link which would reduce maintenance each (major) release.

Reviewer 3: As per my comment below the latest link for which package? Currently its the tarball with "other options" being found out on GitHub. Until we extend the UX on the /downloads page we would need to follow this approach.

@devinbinnie does the tarball install method also support the auto-updater?

Reviewer 4: @amyblais We do not have latest download link for desktop app.

      - 64-bit systems: `mattermost-desktop-5.0.4-linux-x64.tar.gz <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-x64.tar.gz>`__
      - 32-bit systems: `mattermost-desktop-5.0.4-linux-ia32.tar.gz <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-ia32.tar.gz>`__

    2. Extract the archive to a convenient location, then execute ``mattermost-desktop`` located inside the extracted directory.

    3. To create a Desktop launcher, open the file **README.md** and follow the instructions in the **Desktop launcher** section.

    **Ubuntu and Debian-based systems**

>     Unofficial, community-driven ``.deb`` packages are available.

Reviewer: @esethna Maybe you can confirm if we want to add to https://mattermost.com/download/ and what that would look like. I can create a ticket for Marketing team.

>     Unofficial, community-driven ``.deb`` packages are available.

Reviewer 5: @amyblais please check with @nab-77 on this as it's distribution related.

>     Unofficial, community-driven ``.deb`` packages are available.

Reviewer 3: @devinbinnie Does the auto-updater work for all client app versions (Linux tar included)? I ask as there the APT/DEB is an additional Linux client (i.e. not the default).

>     Unofficial, community-driven ``.deb`` packages are available.

Reviewer 2: @nab-77 Unfortunately not, here's the breakdown for you:

Windows: Works for all distributions (EXE, MSI, ZIP)
macOS: No built-in auto-updater functionality due to security issues, so we will encourage users to use the App Store build
Linux: Auto-updates only supported using the AppImage. However, if users install the APT repository, they will get updates when they update their system. We're working on doing the same thing for RPM, but that's a bit far away.

Hope that helps.

>     Unofficial, community-driven ``.deb`` packages are available.

Author: @nab-77 - Any specific action I should take based on @devinbinnie's response within the scope of this Desktop App auto-updater PR?

>     Unofficial, community-driven ``.deb`` packages are available.

Author: @devinbinnie & @nab-77 - do we want to include the following deb repo install commands in the docs?

```
curl -o- https://deb.packages.mattermost.com/setup-repo.sh | sudo bash
sudo apt install mattermost-desktop
```

>     Unofficial, community-driven ``.deb`` packages are available.

Reviewer 3: @devinbinnie That's perfect and really helps re how we document/promote.

@amyblais I will touch base with @cwarnermm as the feedback above does change this a fair amount.

@cwarnermm keep your eyes out for a meeting ;-)

    1. Download the latest version of the Mattermost Desktop App:

>       - 64-bit systems `mattermost-desktop-5.0.4-linux-amd64.deb <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-amd64.deb>`__

Author: Need updated download URLs

Reviewer 2: I think we have a "latest" autolink that will always grab the latest build? Not sure what it is though :P

      - 32-bit systems `mattermost-desktop-5.0.4-linux-i386.deb <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-i386.deb>`__

    2. At the command line, execute one of the following commands depending on the package that you downloaded:

>       - 64-bit systems ``sudo dpkg -i mattermost-desktop-5.0.4-linux-amd64.deb``

Author: Need updated download URLs

      - 32-bit systems ``sudo dpkg -i mattermost-desktop-5.0.4-linux-i386.deb``

    3. Run Mattermost as a desktop app.

    **Linux rpm files (Beta)**

>     Unofficial, community-driven ``.rpm`` packages are available.

Author: Are our rpm packages unofficial today?

Reviewer 2: Yes for now they are.

    1. Download the latest version of the Mattermost Desktop App:

>       - 32-bit systems `mattermost-desktop-5.0.4-linux-i686.rpm <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-i686.rpm>`__

Author: Need updated download URLs

      - 64-bit systems `mattermost-desktop-5.0.4-linux-x86_64.rpm <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-x86_64.rpm>`__

    2. At the command line, execute one of the following commands based on the package you downloaded:

      - 32-bit systems ``sudo rpm -i mattermost-desktop-5.0.4-linux-i686.rpm``
      - 64-bit systems ``sudo rpm -i mattermost-desktop-5.0.4-linux-x86_64.rpm``

    3. Run Mattermost as a desktop app.

    **Arch Linux-based systems**

    To install the Desktop client on Arch Linux, see the `Mattermost page <https://wiki.archlinux.org/index.php/Mattermost>`__ on the Arch Linux wiki.

>     **Snapcraft package**

Author: Do we want to continue documenting a snap option in the product documentation? If so, need updated download URLs

    A snap is available for systems that have Snapcraft installed. Snapcraft is installed by default on Ubuntu 16.04 and later, but for most other Linux distributions you can install it manually. To install Snapcraft, see `Install snapd <https://snapcraft.io/docs/core/install>`__ on the Snapcraft website.

    1. At the command line, execute the following command: ``sudo snap install mattermost-desktop --beta``

    2. Run Mattermost as a desktop app.

  .. tab:: Mac 10.9+

    1. Download the latest version of the Mattermost desktop app:
      

>       - `Intel systems <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-mac-x64.dmg>`__

Author: Need updated download URLs

      - `M1 systems <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-mac-m1.dmg>`__ (Beta)

    2. Double-click the download to open the disk image.

    3. Drag the Mattermost application to the **Applications** folder.

    .. tip::
      

>       `Homebrew <https://brew.sh>`__ users can install with ``brew install --cask mattermost``.

Author: Would following this step download the latest version of the desktop app?

Reviewer 2: I've never done it this way myself, so I'm not sure if it works, but it is no longer the recommended way.

As of May 16th we'll be in the App Store so it would make sense to direct them there. I'll have to get an updated link for you for that once we're in.

Author: Thanks, @devinbinnie. Should we remove this set of instructions from the docs since it's no longer our recommended path?

Reviewer 2: Yes, definitely remove this.

Author: Removed. Thanks!

  .. tab:: Windows 8.1+

>     1. Download the latest version of the Mattermost desktop app: `32/64-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-setup-5.0.4-win.exe>`__

Author: Need latest download URLs

    2. From the **\Downloads** folder, right-click on the file ``mattermost-desktop-setup-5.0.4-win.exe`` and select **Open** to start an installer for the app. Once finished, the Mattermost desktop app opens automatically.

    **MSI Installer and Group Policies (Beta)**

    You can download the latest version of the Mattermost Desktop App MSI installer (Beta):

>     - MSI for `64-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-x64.msi>`__

Author: Need latest download URLs

    - MSI for `32-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-x86.msi>`__

    See the `MSI installer and group policy documentation <https://docs.mattermost.com/install/desktop-msi-gpo.html>`__ for instructions on installing the Mattermost Desktop App via an MSI installer and configuring supported group policies. Changes to group policies require you to restart Mattermost for those changes to take effect.

    The following group policies are available:

>     +----------------------------+-----------------------------------------------------------------------------+---------------------------------+

Author: Any changes needed here?

Reviewer 2: You can also disable auto-updates using `Enable Auto Updater`

Author: @devinbinnie - Can you provide more detail on how to disable auto-updates, please?

Reviewer 2: It's the same procedure that you would do to enable/disable server management, or edit the server list. It's just a boolean switch, where if it's set to `true`, updates are enabled. And if `false`, they are disabled.

Reviewer 2: @cwarnermm So this isn't done in the product, it's generally done by system administrators in the Group Policy Editor. This isn't something that should be changed by the user.

Just to clarify, so we don't presently have any documentation currently with respect to editing the GPO settings? I would have figured we would have since the GPO settings have been around a while. If not, is that something we should be documenting, or is that more on the system administrator side of things?

    | Group policy               | Description                                                                 | Required Mattermost version     |
    +============================+=============================================================================+=================================+
    | Enable Server Management   | If disabled, management of servers in the app settings are disabled.        | 4.3 or later                    |
    +----------------------------+-----------------------------------------------------------------------------+---------------------------------+
    | Default Server List        | Define one or more default, permanent servers.                              | 4.3 or later                    |

>     +----------------------------+-----------------------------------------------------------------------------+---------------------------------+

Reviewer 5: Missing the group policy for auto-updates

Author: Thanks, @esethna! @devinbinnie - Is this content you're able to help develop?

Reviewer 2: Do we not already have documentation around changing Group Policy for the other settings? As per this https://github.com/mattermost/docs/pull/5628#discussion_r869607058 it's the same procedure.

Update the Mattermost Desktop App
---------------------------------

For the best experience, we recommend keeping your Desktop App updated to the latest release. 

.. tabs::

  .. tab:: Linux

    Depending on the distribution you're using, your Desktop App updates automatically when a new release is available.

>     - Ubuntu: A Debian repository is available to configure and install. TBD

Author: @devinbinnie - Can you provide these details when available, please?

Reviewer 2: See above (sorry didn't see this comment yet lol)

  .. tab:: Mac 10.9+

    You have two ways to install the Desktop App, and how you install the app determines whether it updates automatically.
    
    - You can install the Desktop App from the `App Store <https://apps.apple.com/us/app/mattermost/id1257222717>`__. When you install through the App Store, your Desktop App updates automatically when a new release is available.
    - You can `download the Desktop App directly from our GitHub release page <https://github.com/mattermost/desktop/releases>`__. However, when you install the Desktop App this way, you can't manually check for updates, and updates won't be installed automatically.

    **Check your Desktop App version**

    Select **Mattermost > About Mattermost** from the menu bar. This dialog indicates when you've downloaded the app from the App Store.

  .. tab:: Windows 8.1+

    Your Desktop App updates automatically when a new release is available. You can check for updates manually any time.

    **Manually check for updates**

    1. Select the **More** |more-icon-vertical| icon located in the top left corner of the Mattermost window, then select **File > Settings...**.
    2. Select **Check for updates**.
    

>     If a newer release of the Desktop App is available, you're prompted to install it.

Reviewer 5: Add here that auto-updates can be disabled via the GPO setting

  :target: https://mattermost.com/deploy
  :alt: Available for Mattermost Self-Hosted deployments.

.. |more-icon-vertical| image:: ../images/dots-vertical_F01D9.svg
  :alt: Use the More icon in the top left corner to access Mattermost Desktop Apps customization settings.

> The Mattermost Desktop App is available for Windows, Mac, and Linux operating systems. The Desktop App supports all the features of the web experience, plus the following features:

Reviewer 6: ```suggestion
The Mattermost Desktop App is available for Windows, macOS, and Linux. The Desktop App supports all the features of the web experience, plus the following features:
```

- `Connect to multiple Mattermost servers <https://docs.mattermost.com/welcome/manage-desktop-app-server-connections.html>`__ from a single interface, and navigate between servers using keyboard shortcuts.
- `Auto-start Mattermost <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ when a user logs into their machine.
- `Add Mattermost <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ to the Windows Start menu, the Taskbar, the Dock, or the System Tray.

> - (Windows/Mac) `Deep link to the Desktop App <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ via the ``mattermost://`` protocol if the app is already installed.

Reviewer 6: ```suggestion
- (Windows/macOS) `Deep link to the Desktop App <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ via the ``mattermost://`` protocol if the app is already installed.
```

- (Linux) `Set up Desktop Entry <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__ for the application to more easily `integrate into a desktop environment <https://wiki.archlinux.org/index.php/Desktop_entries>`__.

The following additional documentation resources are also available:

- `Desktop App changelog <https://docs.mattermost.com/install/desktop-app-changelog.html>`__
- `Minimum software requirements <https://docs.mattermost.com/install/software-hardware-requirements.html#desktop-apps>`__
- `Configure your Desktop App experience <https://docs.mattermost.com/welcome/customize-desktop-app-experience.html>`__
- `Source code <https://github.com/mattermost/desktop>`__
- `Contributor’s guide <https://developers.mattermost.com/contribute/desktop>`__

Install the Mattermost Desktop App
----------------------------------

> You can `download the Desktop App directly from our Downloads page <https://mattermost.com/download/>`__. You may also use the following installation guides for Linux, Mac, and Windows.

Reviewer 6: ```suggestion
You can `download the Desktop App directly from our Downloads page <https://mattermost.com/download/>`__. You may also use the following installation guides for Windows, macOS, and Linux.
```

.. include:: ../upgrade/upgrading-to-v60.rst
  :start-after: :nosearch:

.. tabs::

  .. tab:: Linux

    **Generic Linux package** 

>     1. Download the latest version of the Mattermost desktop app:

Reviewer 6: ```suggestion
    1. Download the latest version of the Mattermost Desktop App:
```

    1. At the command line, execute the following command: ``sudo snap install mattermost-desktop --beta``

    2. Run Mattermost as a desktop app.

  .. tab:: Mac 10.9+

    1. Download the latest version of the Mattermost desktop app:
      
      - `Intel systems <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-mac-x64.dmg>`__
      - `M1 systems <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-mac-m1.dmg>`__ (Beta)

    2. Double-click the download to open the disk image.

    3. Drag the Mattermost application to the **Applications** folder.

  .. tab:: Windows 8.1+

    1. Download the latest version of the Mattermost desktop app: `32/64-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-setup-5.0.4-win.exe>`__

    2. From the **\Downloads** folder, right-click on the file ``mattermost-desktop-setup-5.0.4-win.exe`` and select **Open** to start an installer for the app. Once finished, the Mattermost desktop app opens automatically.

>     **MSI Installer and Group Policies (Beta)**

Reviewer 6: ```suggestion
    **MSI Installer and group policies (beta)**
```

    - 32-bit systems:
    
      .. code-block:: none
 
        sudo rpm -u mattermost-desktop-5.0.4-linux-i686.rpm

    .. tip:: 
      You can review the current version of your Desktop App by selecting the **More** |more-icon-vertical| icon located in the top left corner of the Mattermost window, then selecting **Help > Version...**.

  .. tab:: Generic Linux

    A beta AppImage distribution of a compressed tarball is available. Automatic app updates aren’t supported. You must update your app manually. 

    1. Download the latest version of the Mattermost Desktop App:

      - 64-bit systems: `mattermost-desktop-5.0.4-linux-x64.tar.gz <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-x64.tar.gz>`__
      - 32-bit systems: `mattermost-desktop-5.0.4-linux-ia32.tar.gz <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-linux-ia32.tar.gz>`__

    2. Extract the archive to a convenient location, then execute ``mattermost-desktop`` located inside the extracted directory.

    3. To create a Desktop launcher, open the file ``README.md``, and follow the instructions in the **Desktop launcher** section.

  .. tab:: macOS

>     MacOS 10.15+ is required. You have two ways to install the Desktop App, and how you install the app determines whether it updates automatically.

Reviewer 7: Small thing, should be "macOS"

Author: Given that this is the first word in a sentence, I've elected to apply sentence case.

    **Install from the App Store**

>     We recommend that you install the Desktop App from the `App Store <https://apps.apple.com/us/app/mattermost/id1257222717>`__. When you install through the App Store, your Desktop App updates automatically when a new release is available.

Reviewer 7: This appears to be a link to the iOS app in the app store, not the macOS app. Let me know if I'm missing something, but I only see boards as a macOS app on the app store (https://apps.apple.com/us/developer/mattermost-inc/id654282837). Did a search for Mattermost on the App Store macOS app on my Mac and only Focalboard came up.

Author: Great catch, @rmonga71! This link takes users to the iOS mobile app, not the Desktop App. I'll replace this link as soon as its available. Thank you!!

    **Install the Mattermost Desktop App**

    1. Download the latest version of the Mattermost desktop app: `32/64-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-setup-5.0.4-win.exe>`__
    2. From the **\Downloads** folder, right-click on the file ``mattermost-desktop-setup-5.0.4-win.exe``, then select **Open** to start an installer for the app. Once finished, the Mattermost desktop app opens automatically.

    **MSI Installer and group policies (beta)**

    You can download the latest version of the Mattermost Desktop App MSI installer (Beta):

    - MSI for `64-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-x64.msi>`__
    - MSI for `32-bit version of Windows <https://releases.mattermost.com/desktop/5.0.4/mattermost-desktop-5.0.4-x86.msi>`__

    The following group policies are available:

    +--------------------------+------------------------------------------------------------+----------------------+------------------------+------------------+

>     | Group policy             | Description                                                | Required Mattermost  | Setting                | State options    |

Reviewer 3: I would change "Required Mattermost version" to "Mattermost Release". I say this as the version is in the table.

    |                          |                                                            | version              |                        |                  |
    +==========================+============================================================+======================+========================+==================+
    | Enable Server Management | If disabled, management of servers in the                  | v4.3 or later        | EnableServerManagement | - Not Configured |   
    |                          | app settings are disabled.                                 |                      |                        | - Enabled        |
    |                          |                                                            |                      |                        | - Disabled       |
    +--------------------------+------------------------------------------------------------+----------------------+------------------------+                  |

>     | Default Server List      | Define one or more default, permanent servers.             | v4.3 or later        | DefaultServerList      |                  |

Reviewer 3: Can we please make each Setting "code".

For further assistance, review the `Troubleshooting forum <https://forum.mattermost.com/c/trouble-shoot>`__ for previously reported errors, or `join the Mattermost user community for troubleshooting help <https://mattermost.com/pl/default-ask-mattermost-community/>`__.

> Report Desktop App Issues

Reviewer 6: ```suggestion
Report Desktop App issues
```
